crates/sui-storage/src/indexes.rs

```diff
@@ -143,64 +151,140 @@ impl IndexStore {
     fn get_transactions_by_object<KeyT: Clone + Serialize + DeserializeOwned + PartialEq>(
         index: &DBMap<(KeyT, TxSequenceNumber), TransactionDigest>,
         object_id: KeyT,
-    ) -> SuiResult<Vec<(TxSequenceNumber, TransactionDigest)>> {
-        Ok(index
-            .iter()
-            .skip_to(&(object_id.clone(), TxSequenceNumber::MIN))?
-            .take_while(|((id, _), _)| *id == object_id)
-            .map(|((_, seq), digest)| (seq, digest))
-            .collect())
+        cursor: TxSequenceNumber,
+        limit: Option<usize>,
+        reverse: bool,
+    ) -> SuiResult<Vec<TransactionDigest>> {
+        Ok(if reverse {
+            let iter = index
+                .iter()
+                .skip_prior_to(&(object_id.clone(), cursor))?
+                .reverse()
+                .take_while(|((id, _), _)| *id == object_id)
+                .map(|(_, digest)| digest);
+            if let Some(limit) = limit {
+                iter.take(limit).collect()
+            } else {
+                iter.collect()
+            }
+        } else {
+            let iter = index
+                .iter()
+                .skip_to(&(object_id.clone(), cursor))?
+                .take_while(|((id, _), _)| *id == object_id)
+                .map(|(_, digest)| digest);
+            if let Some(limit) = limit {
+                iter.take(limit).collect()
+            } else {
+                iter.collect()
+            }
+        })
```

Comment:
can we do sth like this

```suggestion
        let mut iter = index
                .iter()
                .skip_prior_to(&(object_id.clone(), cursor))?;
        if reverse {
                iter = iter.reverse()
        }  
        iter = iter.
                 .take_while(|((id, _), _)| *id == object_id)
                .map(|(_, digest)| digest);
        if let Some(limit) = limit {
                iter = iter.take(limit)
        }
        Ok(iter.collect())
```

Reply:
I tried but because `iter.reverse()` have a different type `RevIter` I can't assign it to `iter`